orchagent/neighorch.cpp

```diff
@@ -354,13 +354,31 @@ void NeighOrch::doTask(Consumer &consumer)
             if (m_syncdNeighbors.find(neighbor_entry) == m_syncdNeighbors.end() || m_syncdNeighbors[neighbor_entry] != mac_address)
             {
                 if (addNeighbor(neighbor_entry, mac_address))
+                {
                     it = consumer.m_toSync.erase(it);
+                }
                 else
+                {
                     it++;
+                    continue;
+                }
             }
             else
+            {
                 /* Duplicate entry */
                 it = consumer.m_toSync.erase(it);
+            }
+
+            /* Remove remaining DEL operation in m_toSync for the same neighbor.
+             * Since DEL operation is supposed to be executed before SET for the same neighbor
+             * A remaining DEL after the SET operation means the DEL operation failed previously and should not be executed anymore
+             */
+            auto it_rem = consumer.m_toSync.find(key);
+            if (it_rem != consumer.m_toSync.end() && kfvOp(it_rem->second) == DEL_COMMAND)
+            {
+                it = consumer.m_toSync.erase(it_rem);
```

Comment:
"it" was already in the right position, we don't need assign it again. 
I also think it is more efficient to travel backwards (it_rem = it--), and check if the key is the same as the current one, if so, just erase the key.